### DISCUSSION TITLE
[Bug]: Inline env variable doesn't work in "execInContainer"

### DISCUSSION THREAD

**Discussion by gredwhite** [testcontainers/testcontainers-java#7654] [Q&A]
### Module

Core

### Testcontainers version

1.18.3

### Using the latest Testcontainers version?

Yes

### Host OS

Windows

### Host Arch

x64

### Docker version

```shell
Docker version 24.0.6, build ed223bc
```

### What happened?

In my docker desktop I execute this command successfully:
```
LDB_MODULES_PATH=/usr/lib/samba/ldb/ ldbmodify -H /var/lib/samba/private/sam.ldb /tmp/test.ldif 
Modified 0 records successfully
```

I want to do the same in testcontainers:
```
SAMBA_IMAGE = "instantlinux/samba-dc:4.18.5-r0"
val sambaContainer = GenericContainer(SAMBA_IMAGE)
....
sambaContainer.execInContainer(
            "LDB_MODULES_PATH=/usr/lib/samba/ldb/",
            "ldbmodify",
            "-H",
           "/var/lib/samba/private/sam.ldb",
            "/tmp/test.ldif",
        )
```
But I get the error:
```
OCI runtime exec failed: exec failed: unable to start container process: exec: "LDB_MODULES_PATH=/usr/lib/samba/ldb/": stat LDB_MODULES_PATH=/usr/lib/samba/ldb/: no such file or directory: unknown
```
I've found the workaround:
```
val sambaContainer = GenericContainer(SAMBA_IMAGE)
                     ...
                    .withEnv("LDB_MODULES_PATH","/usr/lib/samba/ldb/")
sambaContainer.start()
sambaContainer.execInContainer(
                "ldbmodify",
                "-H",
               "/var/lib/samba/private/sam.ldb",
                "/tmp/test.ldif",
            )
```

But I still think that my initial attempt should work.

**Comment by eddumelendez** [comment]
I think you may need something like execInContainer("sh", "-c", "the whole command")

If want to receive more help, please share a reproducible project, it can be a zip file or a github repository.

**Comment by guicamest** [comment]
I had the same issue and found @eddumelendez answer really helpful, thanks a lot!

I think it would be great to mention this in the [Executing commands](https://java.testcontainers.org/features/commands/) page, wdyt?

Besides documentation, maybe it would be nice that the library provides some method for it, something like:
```
execInContainer(Map<String, String> env, String... command)

// or something more elaborated with some kind of exec builder:
// .env() call is optional
exec().env(Map<String, String>).run(String... command)
```

**Reply by eddumelendez** [reply]
See https://github.com/testcontainers/testcontainers-java/pull/7816

**Reply by guicamest** [reply]
Amazing, really like the improvement 💪 Thanks a lot! 🥳
